UnityProject/Assets/Scripts/UI/Right click/Rightclick.cs

```diff
@@ -0,0 +1,155 @@
+			Vector3 position = Camera.main.ScreenToWorldPoint(UnityEngine.Input.mousePosition);
+			position.z = 0f;
+			//gets Items on the position of the mouse
+			List<GameObject> objects = UITileList.GetItemsAtPosition(position);
+			//Generates menus
+			Generate (objects);
+			//Logger.Log ("yo", Category.UI);
+			if (options.Count > 0){
+				RadialMenuSpawner.ins.SpawnRadialMenu(options);
+			}
+
+		}
+		
+	}
+	private void Generate(List<GameObject> objects){
+		options = new List<Menu> ();
+		for (int i = 0; i < objects.Count; i++) {
+			Menu newMenu = new Menu();
+			newMenu.colour = Color.gray;
+			ItemAttributes Nameues = objects[i].GetComponent<ItemAttributes>();
+			if (Nameues == null) {
+				newMenu.title = "Unknown";
+			} else {
+				newMenu.title = Nameues.itemName;
+			}
+
+			SpriteRenderer UseSprite = objects[i].GetComponentInChildren<SpriteRenderer>();
+			if (UseSprite == null) {
+				newMenu.sprite = ins.Spritenames [0];
+			} else {
+				newMenu.sprite = UseSprite.sprite;
+			}
+
+			//Find all monoBehaviours on object and store in a list
+			MonoBehaviour[] scriptComponents = objects[i].GetComponents<MonoBehaviour>();
+
+			//For each monoBehaviour in the list of script components
+			foreach (MonoBehaviour mono in scriptComponents) {
+				Type monoType = mono.GetType();
+				foreach (MethodInfo method in monoType.GetMethods(BindingFlags.Public | BindingFlags.Instance)) {
+					var attributes = method.GetCustomAttributes(typeof(contextMethod), true);
+					if (attributes.Length > 0) {
+						
+						//Logger.Log("Script: " + mono + " Method: " + method.ToString(), Category.UI);
+						//Logger.Log (method.ToString (), Category.UI);
+						Menu NewSubMenu = new Menu();
+						contextMethod contextMethodMenu = (contextMethod)method.GetCustomAttributes (typeof(contextMethod), true)[0];
+						NewSubMenu.colour = Color.gray;
+						NewSubMenu.Item = objects[i];			
+						NewSubMenu.title = contextMethodMenu.ButtonTitle;
+						NewSubMenu.sprite =  SpriteDictionary[contextMethodMenu.SpriteName];
+						NewSubMenu.Mono = mono;
+						NewSubMenu.Method = method;
+
+						newMenu.SubMenus.Add (NewSubMenu);
+
+					}
+				}
+			}
+
+			//for (int L = 0; L < 3; L++) {
+			//	Menu NewSubMenu = new Menu();
+			//	NewSubMenu.colour = Color.gray;
+			//	NewSubMenu.title = "sub " + ins.names[L];
+			//	NewSubMenu.sprite = ins.Spritenames[0];
+			//	newMenu.SubMenus.Add (NewSubMenu);
+			//}
+			//Sort
+			List<Menu> Sortlist = newMenu.SubMenus;
+			Menu[] array = new Menu[MenuOrder.Count+1];
+
+			foreach (Menu SubMenu in Sortlist) {
+				array[MenuOrder[SubMenu.title]] = SubMenu;
+			}
+
+			newMenu.SubMenus = new List<Menu>();
+
+			for (int S = 0; S < array.Length; S++)  {
+				if (!(array[S] == null)) {
+					newMenu.SubMenus.Add (array [S]);
+				}
+			}
+
+			ins.options.Add (newMenu);
+		}
+	}
+}
+
+[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+public class contextMethod : Attribute
```

Comment:
Please rename `contextMethod` to `ContextMethod`, classes are meant to be PascalCase